enhancements/csvless-bundles.md

```diff
@@ -24,209 +24,488 @@ status: implementable
+```
 
-The following are parts of a CSV that are NOT directly inferrable from plain kubernetes resources:
+The `bundle.yaml` file consolidates all of the additional info needed by OLM to manage the life cycle of the operator. The set of additional info is exactly (w/ required info in **bold**):
 
-- Name
-- Descriptors
-- Description
-- Display Name
-- Keywords
-- Labels
-- Links
-- Maintainers
-- Maturity
-- Provider
-- Selector
-- Icon
-- Runtime Metadata
-- MinKubeVersion
-- (Future) Optional APIs
-- Replaces
-- Version
-- InstallModes
+- **name**
+- **version**
+- **minimum kubernetes version**
+- description
+- display name
+- keywords
+- labels
+- links
+- maintainers
+- maturity
+- provider
+- selector
+- icon
+- API descriptions
 
-This non-inferrable data will be gathered through a mixture of resource annotations and an `olm.yaml` file in the `metadata` directory of the bundle.
+The naming of `bundle.yaml` is strict and is used to identify the file. 
 
-Ex. _olm.yaml_
+_Ex. olm.yaml_
 
 ```yaml
-name: packageserver-v1.14.1
-maturity: alpha
-version: 0.14.1
-displayName: Package Server
-description: Represents an Operator package that is available from a given CatalogSource which will resolve to a ClusterServiceVersion.
-minKubeVersion: 1.11.0
-keywords: ['packagemanifests', 'olm', 'packages']
+operators.operatorframework.io.bundle.mediatype: registry+v2
+name: kiali-operator.v1.9.1
+version: 1.9.1
+minKubeVersion: 1.17.3
+description: |
+    ## About this Operator
+
+    ### Kiali Custom Resource Configuration Settings
+
+    For quick descriptions of all the settings you can configure in the Kiali
+    Custom Resource (CR), see the file
+    [kiali_cr.yaml](https://github.com/kiali/kiali/blob/v1.9.1/operator/deploy/kiali/kiali_cr.yaml)
+
+    Note that the Kiali operator can be told to restrict Kiali's access to
+    specific namespaces, or can provide to Kiali cluster-wide access to all
+    namespaces.
+    
+    ...
+displayName: Kiali Operator
+keywords: ['service-mesh', 'observability', 'monitoring', 'maistra', 'istio']
+labels:
+  name: kiali-operator
+  url: https://www.prometheus.io/
 maintainers:
-- name: Red Hat
-  email: openshift-operators@redhat.com
+- name: Kiali Developers Google Group
+  email: kiali-dev@googlegroups.com
+maturity: beta
 provider:
-  name: Red Hat
-links:
-- name: Package Server
-  url: https://github.com/operator-framework/operator-lifecycle-manager/tree/master/pkg/package-server
-installModes:
-- type: OwnNamespace
-  supported: true
-- type: SingleNamespace
-  supported: true
-- type: MultiNamespace
-  supported: true
-- type: AllNamespaces
-  supported: true
-icon: my-icon-png-base64-encoded
-labels:
-  operated-by: packageserver
+  name: Kiali
+selector:
+  matchLabels:
+    name: kiali-operator
+icon:
+  - base64data: PHN2ZyB3a...
+    mediatype: image/svg+xml
+version: 1.9.1
 apis:
-- group: packages.operators.coreos.com
-  version: v1
-  kind: PackageManifest
+- apiVersion: kiali.io/v1alpha1
```

Comment:
Just thinking completely out loud:

Can we define what included here is actually required and what is just for opt-in feature wise?

In this case, I imagine there are definitely some operator authors that just want to drop their CRDs in the `manifests/` directory and be done with it. If there is stuff in here that's required, can we call out why the CRD needs to be explicitly referenced here?

Reply:
I am also curious if the amount of CRD information in `bundle.yaml` can be streamlined.

Reply:
> Can we define what included here is actually required and what is just for opt-in feature wise?

Required fields are in bold above the example, or are we talking about what's required in the `apis` node should it be specified?

> In this case, I imagine there are definitely some operator authors that just want to drop their CRDs in the manifests/ directory and be done with it.

This is opt-in.

> can we call out why the CRD needs to be explicitly referenced here?

It's more like just tagging an API with a descriptor. Does that make things more clear?

Reply:
> I wonder if making these their own files may make it more clear also easier to find, reason about, and generate?

So we're talking about having a `descriptors.yaml` file as well as `bundle.yaml`, right?

Reply:
I was hoping that we could find a way to add this to the CRD. I imagine that it would be easier for the UI to read that data that way. @spadgett can you look over?